### PR TITLE
Convert tests to PHPUnit

### PR DESCRIPTION
**Pull request by ricordisamoa**
* Add development dependency on phpunit/phpunit ^6.0
* Update composer.lock for PHPUnit only
* Change .travis.yml to use PHPUnit in vendor directory
* Put expected test results into a new directory

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -18,5 +18,11 @@
     "phpdocumentor/reflection-common": "^1.0@dev",
     "phpdocumentor/reflection-docblock": "^3.0@dev",
     "sebastian/diff": "^1.3"
+  },
+  "require-dev": {
+    "phpunit/phpunit": "^6.0"
+  },
+  "scripts": {
```
Comment (dunglas) [inline composer.json]: Why not running `phpunit` directly?

**Review by dunglas — CHANGES_REQUESTED** [review]

**Comment by dunglas** [comment]
I just left a minor comment but it's awesome!! Thank you very much.

File: composer.json (hunk 1 above)
Comment (ricordisamoa) [reply composer.json]: Adding the Composer script allows to easily run the correct version of PHPUnit (but also other tools such as PHPCS, should they be added in the future) without installing anything globally or modifying the PATH. See: https://getcomposer.org/doc/articles/scripts.md#writing-custom-commands
Do you want me to drop the script? (In that case, would `.travis.yml` require the use of `vendor/bin/phpunit` to work with PHPUnit 6? I'm not sure)

**Comment by ricordisamoa** [comment]
Thank you for this tool and for the quick response.

File: composer.json (hunk 1 above)
Comment (dunglas) [reply composer.json]: Yes I prefer this way (it doesn't pollute the `composer.json` file with CI related commands).

Comment (ricordisamoa) [reply composer.json]: It should be fine now.

**Review by dunglas — APPROVED** [review]

**Comment by dunglas** [comment]
Thank you @ricordisamoa

**Comment by ricordisamoa** [comment]
Thanks

**Pull request merged**